### PR TITLE
Add script and workflow to clean caches connected to community extensions

### PR DESCRIPTION
**Pull request by carlopi**
Example to try the script:
```
bash ./scripts/clean_caches.sh duckdb-community-extensions community-extensions "*/quack.duckdb_extension.*"
```

Script is then invoked by clean_caches.yml, that can be generalized further to act as middle-man also from other cases.

### CODE REVIEW

File: .github/workflows/clean_caches.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#
+# Reusable workflow that deploys the artifacts produced by .github/workflows/_extension_distribution.yml
```
Comment (samansmink) [inline .github/workflows/clean_caches.yml:2]: comment is copied over

File: .github/workflows/clean_caches.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+on:
+  workflow_call:
+    inputs:
+      # The name of the extension
+      pattern:
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      # The name of the extension
```
Comment (samansmink) [inline .github/workflows/clean_caches.yml:18]: nit: comment is outdated

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -57,4 +57,10 @@ jobs:
...
       ref: ${{ matrix.COMMUNITY_EXTENSION_REF }}
 
-
+  clean_cache:
+     needs:
+       - deploy
+     uses: ./.github/workflows/clean_caches.yml
+     secrets: inherit
+     with:
+       pattern: '*${{ matrix.COMMUNITY_EXTENSION_NAME }}.duckdb_extension*'
```
Comment (samansmink) [inline .github/workflows/deploy.yml:66]: this matches too much i think? extension names now match every extension ending in the extension name

Comment (carlopi) [reply .github/workflows/deploy.yml]: Improved, thanks!

**Pull request merged**